xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-sources/src/main/java/org/xwiki/notifications/sources/internal/UsersParameterHandler.java

```diff
@@ -90,6 +102,24 @@ public void handleUsersParameter(String users, NotificationParameters parameters
             parameters.filters.add(new FollowedUserOnlyEventFilter(entityReferenceSerializer, userList));
 
             addFilterPreference(parameters, userList);
+        } else if (parameters.user != null) {
+            // if we have a user (but no "users") then we should also display personal messages from followed users.
+            // the other types of messages get included, but for personal messages the filter needs
+            // a matching filter preference so we loop though preferences to see if they have
+            // a preference for this (using a copy to guard against unwanted modifications)
+            try {
+                for (NotificationFilterPreference filterPref
```

Comment:
I think you could break the for whenever the filter is found. Would be a small perf improvment.

Reply:
I don't think that would work, since IIUC, the goal is to copy the filters for each followed users. Adding a break would only add a filter for the first followed user found, and ignore all the others.

Reply:
>  the goal is to copy the filters for each followed users

no, you only get the filter for the user in `parameters.user`: that's the filter job to check if the message comes from the followed users.

Reply:
I agree with the idea but that's not how it is working currently. If `U1` follows `U2` and `U3`, `notificationFilterPreferenceManager.getFilterPreferences(parameters.user)` will return 2 filterers with a filter name equals to `EventUserFilter.FILTER_NAME`, where the owner is U1 and the user is the followed user.

AFAIK we don't have a generic "followed users" filter.

Reply:
hmmm right. So indeed what I proposed was a bad idea. I guess it would deserve a comment.

Reply:
I added another sentence to the comment block above. I hope that clarifies things.